src/ros_filter.cpp

```diff
@@ -736,7 +736,7 @@ void RosFilter<T>::integrateMeasurements(const rclcpp::Time & current_time)
 
     // If we get a large delta, then continuously predict until
     if (last_update_delta >= filter_.getSensorTimeout()) {
-      predict_to_current_time = true;
+      predict_to_current_time_ = true;
```

Comment:
So the existing logic is [the same in ROS 1](https://github.com/cra-ros-pkg/robot_localization/blob/bf02bfbe5065071a143ffebf0cdc4fc3c677b1eb/src/ros_filter.cpp#L604), and that works absolutely fine, even at very high frequencies.

The aim of this logic is:

- If the user has `predict_to_current_time` set to `true`, we always predict to the current time
- If there is a large time delay between measurements, then _just for that cycle_, we predict to the current time so that our state estimate doesn't time out

Your change breaks this logic: if the user's sensor data ever experiences a large gap, it will permanently set `predict_to_current_time_` to `true`.

Is there a reason you can't just set `predict_to_current_time` to true in your configuration?

Reply:
> as here the filter will slow down too much from 100 hz to somehow 10 hz and in somecases completely stuck

Do you have a firm understanding of why that would happen? Do you have a bag that can replicate this behavior? Something tells me we're papering over the real issue with this change. I've run the ROS 1 version at 100 Hz with 5 IMUs reporting at the same rate, and have never had so much as a hiccup. We really need to provide data that can replicate the issue. Otherwise, I'm reluctant to accept that the filter would completely stop unless I have the exact code path that would lead to that condition.

> The filter continue always predict/update cycles and integrating the incoming measurements as it should, and the filter is kind of always in ready state to keep projecting the state forward when meas delay or no meas for sometime, rather than keep switching to do it for one cycle or not depending on measurements that have random delays compared to the estimation cycle time period.

No, this _is_ a problem. There are many applications where we only want the filter to produce output when it receives measurement, and not use the kinematic model to predict to the current time stamp.

The bottom line is that this changes the filter behavior for a very large number of users, just to suit your use case. If you want to _add_ a parameter called `predict_to_current_latch` or something (along with the relevant documentation), that would be fine. Then you can change the logic above to be something like:

```
  bool predict_to_current_time = predict_to_current_time_;

...

    if (last_update_delta >= filter_.getSensorTimeout()) {
      predict_to_current_time = true;
      
      if (predict_to_current_time_latch_)
      {
        predict_to_current_time_ = true;
      }
```